### ISSUE TITLE
Non-symmetric sparsity pattern from graphical lasso

### ISSUE THREAD

**Issue by hoangtt1989** [zdk123/SpiecEasi#10]
Hello,

I've been using the `glasso` method in `SpiecEasi` for OTU count data. I used the following code to generate the optimal precision matrix, etc.

`se_gl <- spiec.easi(phy_sub, method = 'glasso', lambda.min.ratio = 1e-2, nlambda = 30,       icov.select.params=list(rep.num=50, ncores = detectCores()))`

Where `phy_sub` is a `phyloseq` object. However, I found that when I check `isSymmetric(se_gl$refit)` (from my understanding this would specify the edges for this undirected graph, which should be symmetric) I get `FALSE`. Do you have any idea why this might be the case, or am I misunderstanding the adjacency matrix?

**Comment by zdk123** [comment]
You are correct that this is unexpected behavior. The error is coming from `huge` and I've seen it once before due to, I expect, numerical issues that result in consistencies for one of the edges.

Can you check the path of adjacency matrices or icov matrices (around the stars-selected index) to see how large the deviation is between the upper/lower entries when both values are present? From there you can decide how you want to symmetrize the refit matrix.

In a future release, I will include this as a automatic check.

Thanks!

**Comment by hoangtt1989** [comment]
I looked at `se_gl$opt.icov - t(se_gl$opt.icov` and found `max(abs())` of the entries in the result and found `0.0002917454`. I also looked at `se_gl$refit - t(se_gl$refit)` and found that 6 of the `colSums` for this result were non-zero.

**Comment by rec3141** [comment]
FYI I'm getting this error as well

```
> dim(all.small.se)
[1]  268 1122

> se.gl.small <- spiec.easi(all.small.se, method='glasso', lambda.min.ratio=1e-2, nlambda=20, icov.select.params=list(rep.num=50,ncores=4))
Normalizing/clr transformation of data with pseudocount ...
Inverse Covariance Estimation with glasso ...
Model selection with stars ...
Done!

> ig.gl <- adj2igraph(se.gl.small$refit)

Error in graph.adjacency.sparse(adjmatrix, mode = mode, weighted = weighted,  : 
  Please supply a symmetric matrix if you want to create a weighted graph with mode=UNDIRECTED.

> max(abs(se.gl.small$opt.icov - t(se.gl.small$opt.icov)))
[1] 0.000421661

```

**Comment by zdk123** [comment]
Sorry for the extreme delay in addressing this issue. It has been fixed in 599a034 in the `dev` branch. I will close once a PR is finalized to master.

**Comment by zdk123** [comment]
fixed in dev.
